[VP][DAGCombine] Use `simplifySelect` when combining vp.select.

AinsleySnow opened the pull request:
Hi all,

This patch is a follow-up of #79101. It migrates logic from `visitVSELECT` to `visitVP_SELECT` to simplify `vp.select`. With this patch we can do the following combinations:

```
vp.select undef, T, F --> T (if T is a constant), F otherwise
vp.select <condition>, undef, F --> F
vp.select <condition>, T, undef --> T
vp.select false, T, F --> F
vp.select <condition>, T, T --> T
```

I'm a total newbie to llvm and I'm sure there's room for improvements in this patch. Please let me know if you have any advice. Thank you in advance!

arsenm commented:
Are you sure these tests are hitting the DAG combine, and not getting cleaned up in InstSimplifyO

AinsleySnow committed:
[DAGCombine] Add test for folding vp.select.

AinsleySnow committed:
[VP][DAGCombine] Use simplifySelect when combining vp.select.

AinsleySnow commented:
I addressed the comments in the newly-pushed commits.

I added pre-commit test cases to demonstrate that the tests will hit the DAG combine. Before we use `simplifySelect`, there are redundant `vmand`/`vmor` in the generated assembly code but after that they turn into single `vmv1r.v`s.

Thank you for your suggestions!

arsenm reviewed (approved):
I think this shows we have missing handling in InstSimplify too. These shouldn't have reached the combiner

AinsleySnow commented:
Thank you. We can add logic to InstCombine in later patches.

By the way could you please help me commit these changes? I have no write access right now.
